packages/zoltan2/src/util/Zoltan2_ImbalanceMetricsUtility.hpp

```diff
@@ -156,14 +156,16 @@ template <typename scalar_t, typename lno_t, typename part_t>
                                              // [0,maxPartPlusOne)
 
   part_t globalSumSize = maxPartPlusOne * numMetrics;
-  scalar_t * sumBuf = new scalar_t [globalSumSize];
+
+  typedef scalar_t temp_t; // TODO remove this - added to resolve cuda warnings I didn't understand yet
```

Comment:
I removed this. It causes a warning now but this is common in many spots:
warning: non-constant array new length must be specified without parentheses around the type-id [-Wvla].
Seems nvcc things there are parentheses around the type scalar_t but if I make a local typedef it avoids the warning.

Reply:
Teuchos::SerialDenseMatrix has this issue too.  I don't think we need to worry about it, but I would prefer to see `std::unique_ptr` used here instead of explicit `new` and `delete`.